graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/PropertySetter.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.internal
+
+import com.netflix.graphql.dgs.exceptions.DgsInvalidInputArgumentException
+import org.springframework.beans.PropertyAccessor
+import org.springframework.beans.PropertyAccessorFactory
+import org.springframework.core.convert.ConversionException
+import org.springframework.core.convert.support.DefaultConversionService
+
+/**
+ * <code>PropertySetter</code> allows for calling setter methods when they exist and falls back to direct field access
+ * when they don't.
+ * <p>
+ * Type conversion is done explicitly as there are some cases that Spring <code>PropertyAccessor</code>s don't handle
+ * out of the box.
+ */
+class PropertySetter(target: Any, private val conversionService: DefaultConversionService) {
```

Comment:
I am not sure I would bother with a separate class here, I would suggest just inlining the new behavior into `DefaultInputObjectMapper`.

1. Both methods in `PropertyAccessorFactory` return a `ConfigurablePropertyAccessor`, which has a method to configure the conversion service to use. If we just use that, it's no longer necessary to do any explicit conversion ourselves, so the convertValue method goes away, and `trySet` basically isn't needed either.
2. One less extra class that needs to be allocated, although in and of itself maybe not a huge concern, I am mostly not sure the extra abstraction is warranted.

If we combine the above two, I think the change might be something like this:
```kotlin
if (setterAccessor.isWritableProperty(name)) {
  setterAccessor.setPropertyValue(name, value)
} else if (fieldAccessor.isWritableProperty(name)) {
  fieldAccessor.setPropertyValue(name, value)
} else {
   nrOfFieldErrors++
  logger.warn("Field '{}' was not found on Input object of type '{}'", name, targetClass)
  continue
}
```
The above also avoids calling `isWritableProperty` twice on the property / setter accessor.

Reply:
I like when this kind of fallback logic is encapsulated/abstracted away, but it might be a matter of preference/style, and I won't argue about it!